### PR TITLE
#166719067 Write unit tests for the category routes

### PR DESCRIPTION
**Pull request by Chiagozielam**
#### What does this PR do?
Write unit tests for the category routes

#### Description of Task to be completed?
- create a new sub folder in `category` with name `_test_`
- create a file with name `category.spec.js`
- write tests for the category routes

#### How should this be manually tested?
in your terminal try `npm run test` and see if the tests are passing

#### What are the relevant pivotal tracker stories?
[#166719067](https://www.pivotaltracker.com/story/show/)

#### Questions:

### CODE REVIEW

File: src/app/category/_tests_/category.spec.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        name: 'bloom',
+      };
+      const user = await createUser(newUserRequestObject);
+      const response = await addCategory(requestObject, user.body.token);
+      const errorMessage = 'You must have role: [admin] to access this route';
+      expect(response.body.error).to.equal(errorMessage);
+      expect(response.status).to.equal(401);
+    });
+
+    //PASSING
```
[the remark this answers is not in the document]
Comment (ebenezerdon) [reply src/app/category/_tests_/category.spec.js]: @Chiagozielam, can you remove your comment from this line?

File: src/app/category/_tests_/category.spec.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+      };
+      const response = await addCategory(requestObject, adminToken);
+      const errorMessage = 'name is required';
+      expect(response.body.error).to.equal(errorMessage);
+      expect(response.status).to.equal(400);
+    });
+
+    //  FIXED THIS ERROR
+
+    it('admin should not be able to add category with a non-string name field ', async () => {
```
[the remark this answers is not in the document]
Comment (ebenezerdon) [reply src/app/category/_tests_/category.spec.js]: Add `// eslint-disable-next-line max-len` before this block.

File: src/app/category/_tests_/category.spec.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+      };
+      const user = await createUser(newUserRequestObject);
+      const response = await addCategory(requestObject, user.body.token);
+      const errorMessage = 'You must have role: [admin] to access this route';
+      expect(response.body.error).to.equal(errorMessage);
+      expect(response.status).to.equal(401);
+    });
+
+    //PASSING
+    it('admin should not be able to add category without the name field', async () => {
```
[the remark this answers is not in the document]
Comment (ebenezerdon) [reply src/app/category/_tests_/category.spec.js]: Let's also add `// eslint-disable-next-line max-len` before this block.

File: src/app/category/_tests_/category.spec.js [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+        ...categoryRequestObject,
+        name: undefined,
+      };
+      const response = await addCategory(requestObject, adminToken);
+      const errorMessage = 'name is required';
+      expect(response.body.error).to.equal(errorMessage);
+      expect(response.status).to.equal(400);
+    });
+
+    //  FIXED THIS ERROR
```
Comment (ebenezerdon) [inline src/app/category/_tests_/category.spec.js]: @Chiagozielam, looks like you forgot this.

Comment (Chiagozielam) [reply src/app/category/_tests_/category.spec.js]: yeah

Comment (Chiagozielam) [reply src/app/category/_tests_/category.spec.js]: Yes, thanks for pointing this out

**Pull request merged**